packages/plugins/plugin-thread/src/ThreadPlugin.tsx

```diff
@@ -310,6 +311,12 @@ export const ThreadPlugin = (): PluginDefinition<ThreadPluginProvides> => {
             case 'article':
             case 'complementary': {
               const location = navigationPlugin?.provides.location;
+              const { plugins } = usePlugins();
+
+              // TODO(Zan): This feels a bit clunky.
```

Comment:
not significantly better but you can factor this logic out and use `useResolvePlugin` instead